src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java

```diff
@@ -1290,7 +1319,13 @@ public KinesisClientLibConfiguration withMaxCacheByteSize(final int maxCacheByte
     }
 
     public KinesisClientLibConfiguration withDataFetchingStrategy(String dataFetchingStrategy) {
-        recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.valueOf(dataFetchingStrategy));
+        switch (dataFetchingStrategy.toUpperCase()) {
```

Comment:
You should be able to use DataFetchingStrategy.valueOf(dataFetchingStrategy) to get the enum value or throw an exception.